turn/src/allocation/mod.rs

```diff
@@ -328,14 +328,29 @@ impl Allocation {
             let mut buffer = vec![0u8; RTP_MTU];
 
             loop {
-                let (n, src_addr) = match relay_socket.recv_from(&mut buffer).await {
-                    Ok((n, src_addr)) => (n, src_addr),
-                    Err(_) => {
-                        if let Some(allocs) = &allocations {
-                            let mut alls = allocs.lock().await;
-                            alls.remove(&five_tuple);
+                let timeout = tokio::time::sleep(Duration::from_secs(10));
+                tokio::pin!(timeout);
+
+                let (n, src_addr) = tokio::select! {
+                    _ = &mut timeout =>{
```

Comment:
can you format the code?

Reply:
I have fixed the format.
I wonder if there is a better way to determine the allocation is closed?

Reply:
Ideally we have something like a [CancelationToken](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html) that fires when some type is dropped or explicitly shutdown instead of relying on timeout. 

Generally for spawned tasks we want have the task stop automatically when the type relates to it is stopped or dropped

Reply:
cc @clia in case you missed this conversation. Can you explore?

Reply:
I'll look into this later when I have time, if you haven't fixed it. Thanks for replies!

Reply:
@melekes @k0nserv I have rewritten the implementation, listening to the drop event and handle it. I have tested on my server and assumed its  effectiveness. Can you guys check these codes for me?